directory/directory_dest.go

```diff
@@ -4,19 +4,55 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 
 	"github.com/containers/image/types"
 	"github.com/opencontainers/go-digest"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
+const version = "Directory Transport Version: 1.0"
+
 type dirImageDestination struct {
-	ref dirReference
+	ref      dirReference
+	compress bool
 }
 
-// newImageDestination returns an ImageDestination for writing to an existing directory.
-func newImageDestination(ref dirReference) types.ImageDestination {
-	return &dirImageDestination{ref}
+// newImageDestination returns an ImageDestination for writing to a directory.
+func newImageDestination(ref dirReference, compress bool) (types.ImageDestination, error) {
+	d := &dirImageDestination{ref: ref, compress: compress}
+
+	// If directory exists check if it is empty
+	// if not empty, check whether the contents match that of a container image directory and overwrite the contents
+	// if the contents don't match throw an error
+	if dirExists(d.ref.resolvedPath) {
+		isEmpty, err := isDirEmpty(d.ref.resolvedPath)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err2 := os.Stat(filepath.Join(d.ref.resolvedPath, "manifest.json"))
+
+		if !isEmpty && err2 != nil && os.IsNotExist(err2) {
+			return nil, errors.Errorf("%q is not empty and not a container image directory. Don't want to overwrite important data, please specifiy a different directory path", d.ref.path)
+		}
+		if err = removeDirContents(d.ref.resolvedPath); err != nil {
+			return nil, errors.Wrapf(err, "error erasing contents in %q", d.ref.resolvedPath)
+		}
+		logrus.Infof("overwriting existing container image directory %q", d.ref.resolvedPath)
+	} else {
+		// create directory if it doesn't exist
+		if err := os.MkdirAll(d.ref.resolvedPath, 0755); err != nil {
+			return nil, err
```

Comment:
Wrap this error with some useful information line the directory it was not able to create.

Reply:
fixed